entries/git-workflow-guidelines/branching.md

```diff
@@ -0,0 +1,146 @@
+
+![Example of simple branching (without GitHub flow).](../images/branching-simple.png)
+
+In this way, branches provide isolation during the development of new features, bug fixes, or experiments. Branching enables parallel development by allowing multiple developers to work on different parts of a project simultaneously without interfering with each other's work. This speeds up development and reduces conflicts that may arise when multiple people modify the codebase at the same time.
+
+![Example of parallel development using branching. Dev = Developer. PR = Pull request. \n NB: Pull requests and reviews process are covered in the post on [pull requests](prs.md).](../images/branching-parallel-development.png)
+
+In the figure above, showing an example of parallel development, two branches are created from the main branch by two developers: bugfix/fix-data-cleaning-error created by Developer 1 and feature/add-variance-plot created by Developer 2. This branching allows for each developer to work on their changes in isolation before creating a pull request and merging the changes into the main branch.
+
+Hopefully, this gave you an intuition about what a branch is and what the benefits of branching are. If you would like more information on branches, check out the [GitHub Documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches) and the [Git Documentation Book](https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell).
+
+## When To Create Branches (and How Much They Should Cover)
+
+Every time you want to add something new to or modify existing work in a Git repository, you should create a new branch (following [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)). Importantly, each branch should represent a specific task or development effort. I.e., a branch should be created for one particular purpose, independently of how many new lines of work is required for that purpose.
+
+If you create branches for *one* particular purpose, it benefits your (and your collaborators') workflow in (at least) two ways:
+
+1. It enables you to give the branch a short, descriptive name that clearly communicates to your collaborators what work has been done on this particular branch.
+2. Smaller changes allows for a faster reviewing and merging to the main branch, since it is easier for your collaborators to get an overview of the work you have done.
+
+As an example, a branch could cover a fix to a particular bug or [issue](issues.md) or add a new feature, such as a new plot, like in the figure showing parallel development above.
```

Comment:
First mention of issues. When should they be "introduced"?

Reply:
Hmm...  Isn't a reference to a file that details them specifically a sort of intro?  Or have I misunderstood something with regards to the link?